tests/integration/security/rbac_v1beta1_test.go

```diff
@@ -738,14 +738,14 @@ func TestV1beta1_TCP(t *testing.T) {
 				}).
 				BuildOrFail(t)
 
-			newTestCase := func(from, target echo.Instance, port string, expectAllowed bool) rbacUtil.TestCase {
+			newTestCase := func(from, target echo.Instance, port string, expectAllowed bool, scheme scheme.Instance) rbacUtil.TestCase {
```

Comment:
I feel it's dangerous to rely on the scheme to send TCP traffic because if the port is declared as HTTP via `--port`, it will still be handled by the HTTP filter and not the TCP filter. Here it works because the port is not declared in the service and using the TCP pass through filter chain, but I think we should make this explicit in the test case that the port is using `--tcp` to always use TCP traffic.

Reply:
scheme filed is used to determine the traffic sent from the client. 

>I feel it's dangerous to rely on the scheme to send TCP traffic because if the port is declared as HTTP via --port, it will still be handled by the HTTP filter and not the TCP filter.

It's incorrect, the `--port` or `--tcp` only determines which kind of protocol is serving port. HTTP filter or TCP filter is determined by the port name. It's possible that user define port name as "http" and protocol as "TCP", the result is the server will listener on the port with TCP protocol but the Envoy configuration will HTTP filter chain.

Reply:
> It's incorrect, the --port or --tcp only determines which kind of protocol is serving port. HTTP filter or TCP filter is determined by the port name.

This is exactly my concern here, the `scheme` is only used to decide the protocol used by the client side, so how do you know the request is going to be handled by the HTTP filter in the server side?

Reply:
I think we can infer if the request is handled by the HTTP filter or not from the response coming from he server as the integration test. We can ensure that by the check in the client side. e.g., you have rbac filter to deny all requests in TCP filter chain and if the request succeeds, the request is not handled by the TCP filter. I think we should test the behavior without knowing the exact details of implementation.